lib/linguist/heuristics.yml

```
@@ -621,6 +621,10 @@ disambiguations:
     pattern: '^\s*(?:use\s+v6\b|\bmodule\b|\bmy\s+class\b)'
   - language: Turing
     pattern: '^\s*%[ \t]+|^\s*var\s+\w+(\s*:\s*\w+)?\s*:=\s*\w+'
+- extensions: ['.tag']
+  rules:
+  - language: Java Server Pages
+    pattern: '\s*<%@.?(taglib|tag|include|attribute|variable)\s'
```

```suggestion
    pattern: '<%[@!=\s]?\s*(taglib|tag|include|attribute|variable)\s'
```

1. `\s*` here is redundant, since the rest of the heuristic will match with or without leading whitespace.

2. Based on my reading of JSP's [syntax reference](https://www.oracle.com/technetwork/java/syntaxref12-149806.pdf), it appears as if `<@= … =>`, `<@! … =>` and `<@ … >` are also supported.

3. `.?` will match virtually anything, including `<%@etag` or `<%@>include`.